File: test_data/code/java/1598-crawler-log-folder.java
class Solution {
    public int minOperations(String[] logs) {
        int depth = 0;
        int minOps = 0;
        for (String log : logs) {
            if (log.equals("../")) {
                depth = Math.max(0, depth - 1);
                minOps += depth == 0 ? 0 : 1;
            } else if (!log.equals("./")) {
                depth++;
            }
        }
        return minOps;
    }
}

File: test_data/code/python/1598-crawler-log-folder.py
from typing import List

class Solution:
    def minOperations(self, logs: List[str]) -> int:
        stack = []
        res = 0
        for log in logs:
            if log == "../":
                if stack:
                    stack.pop()
            elif log != "./":
                stack.append(log[1:])
            else:
                continue
        return res + len(stack)